ast/term.go

```diff
@@ -1810,6 +1816,161 @@ func ObjectTerm(o ...[2]*Term) *Term {
 	return &Term{Value: NewObject(o...)}
 }
 
+func LazyObject(blob map[string]interface{}) Object {
```

Comment:
I'm open to suggestions for better names than LazyObject here. Candidates: ImmutableObject, StoreObject, DataObject... 🤔

Reply:
We're delaying the _conversion to `ast.Value`_ for the object's keys and values. But putting off some work sure sounds "lazy" to me. Although I could be convinced to call it `ProcrastinatingObject` instead...